drivers/azure/azure.go

```diff
@@ -329,7 +332,12 @@ func (d *Driver) Stop() error {
 
 	log.Debugf("stopping %s", d.MachineName)
 
-	return vmClient.ShutdownRole(d.MachineName, d.MachineName, d.MachineName)
+	if err := vmClient.ShutdownRole(d.MachineName, d.MachineName, d.MachineName); err != nil {
+		return err
+	}
+
+	d.IPAddress = ""
```

Comment:
I didn't want to risk leaving the machine in an IP-less state if `vmClient.ShutdownRole` errors.